server/routers/activityRecord.ts

```diff
@@ -0,0 +1,85 @@
+import { TRPCError } from '@trpc/server';
+import { z } from 'zod';
+
+import { createTRPCRouter, privateProcedure } from '../trpc';
+
+export const activityRecordRouter = createTRPCRouter({
+  list: privateProcedure.query(async ({ ctx }) => {
+    const activityRecords = await ctx.prisma.activityRecord.findMany({
+      where: {
+        userId: ctx.user?.id,
+      },
+    });
+
+    return activityRecords;
+  }),
+
+  create: privateProcedure
+    .input(z.object({ activityId: z.number(), addedAmount: z.number() }))
+    .mutation(async ({ input, ctx }) => {
+      const newActivityRecord = await ctx.prisma.activityRecord.create({
+        data: {
+          activityId: input.activityId,
+          userId: ctx.user?.id,
+          addedAmount: input.addedAmount,
+        },
+      });
+
+      return newActivityRecord;
+    }),
+
+  read: privateProcedure
+    .input(z.object({ id: z.number() }))
+    .query(async ({ input, ctx }) => {
+      const activityRecord = await ctx.prisma.activityRecord.findUnique({
+        where: {
+          id: input.id,
+        },
+      });
+
+      if (ctx.user?.id !== activityRecord?.userId) {
+        throw new TRPCError({
+          code: 'FORBIDDEN',
+          message: 'No access to this resource',
+        });
+      }
+
+      return activityRecord;
+    }),
+
+  update: privateProcedure
+    .input(
+      z.object({
+        id: z.number(),
+        activityId: z.number(),
+        date: z.date(),
+        addedAmount: z.number(),
+      }),
+    )
+    .mutation(async ({ input, ctx }) => {
+      const updatedActivityRecord = await ctx.prisma.activityRecord.update({
+        where: {
+          id: input.id,
+        },
+        data: {
+          activityId: input.activityId,
+          date: input.date,
+          addedAmount: +input.addedAmount,
+        },
+      });
+
+      return updatedActivityRecord;
+    }),
+
+  delete: privateProcedure
+    .input(z.object({ id: z.number() }))
+    .mutation(async ({ input, ctx }) => {
+      const deletedActivityRecord = await ctx.prisma.activityRecord.delete({
+        where: {
+          id: input.id,
+        },
+      });
+
+      return deletedActivityRecord;
```

Comment:
You are missing the "forbidden" access check. So right meow, I could be updating and deleting your activity, with my JWT token.

So, let's unify the approach here. Either go with another where param, which would be `userId: ctx.userId` => the userId from JWT token. 

Or you can do the forbidden as above. But choose the same for all.

I think the "forbidden" is slightly better cause it gives more info, even if it is more boilerplate-ish, and it could somehow expose the existence of some user (that the userId exists), but I don't think we need to worry about it here :D (by here I don't mean this project, but the fact it is some random resource, the fact somebody would already have some user's activityId would mean he probably has is userId as well)

Reply:
i will unify it by the previous example